### PR TITLE
SpecHelper: Process body attributes

### PR DESCRIPTION
**Pull request by Swapnilr1**
As per [this discussion](https://github.com/w3c/respec/pull/2188#issuecomment-477266796), `decorateBody` will now process `bodyAttrs`.

I was thinking whether it would be better to replace `intoAttributes()` with a general form that can process both `htmlAttrs` and `bodyAttrs`? :thinking:

### CODE REVIEW

File: tests/spec/SpecHelper.js [hunk 1]
```diff
@@ -96,8 +96,17 @@ function decorateDocument(doc, opts) {
...
 
-  function decorateBody({ abstract = "<p>test abstract</p>", body = "" }) {
+  function decorateBody({
+    abstract = "<p>test abstract</p>",
+    body = "",
+    bodyAttrs,
+  }) {
     doc.body.innerHTML += `<section id='abstract'>${abstract}</section>${body}`;
+    if (bodyAttrs) {
+      Object.keys(bodyAttrs).forEach(key => {
```
Comment (marcoscaceres) [inline tests/spec/SpecHelper.js]:
```suggestion
      Object.entries(bodyAttrs).forEach(([key, value]) => {
```

File: tests/spec/SpecHelper.js [hunk 2]
```diff
@@ -96,8 +96,17 @@ function decorateDocument(doc, opts) {
...
+        doc.body.setAttribute(key, bodyAttrs[key]);
```
Comment (marcoscaceres) [inline tests/spec/SpecHelper.js]:
```suggestion
        doc.body.setAttribute(key, value);
```

File: tests/spec/SpecHelper.js [hunk 3]
```diff
@@ -96,8 +96,17 @@ function decorateDocument(doc, opts) {
     }
   }
 
-  function decorateBody({ abstract = "<p>test abstract</p>", body = "" }) {
+  function decorateBody({
+    abstract = "<p>test abstract</p>",
+    body = "",
+    bodyAttrs,
```
Comment (marcoscaceres) [inline tests/spec/SpecHelper.js]:
```suggestion
    bodyAttrs = {},
```

File: tests/spec/SpecHelper.js [hunk 4]
```diff
@@ -96,8 +96,17 @@ function decorateDocument(doc, opts) {
...
+  }) {
     doc.body.innerHTML += `<section id='abstract'>${abstract}</section>${body}`;
+    if (bodyAttrs) {
```
Comment (marcoscaceres) [inline tests/spec/SpecHelper.js]: If we set the default is set above, we wont need this if statement.

**Comment by marcoscaceres** [comment]
> I was thinking whether it would be better to replace intoAttributes() with a general form that can process both htmlAttrs and bodyAttrs?

I think it’s a good suggestion... but this has only come up once in years, so don’t think it’s worth the effort.

**Comment by marcoscaceres** [comment]
Travis is being painfully slow today :(

**Comment by marcoscaceres** [comment]
Restarted TravisCI

**Comment by Swapnilr1** [comment]
Travis is currently [experiencing issues](https://www.traviscistatus.com/), so I guess we can only wait for tests to run, for hours. :frowning_face:

**Comment by marcoscaceres** [comment]
No biggy, just use this with the other PR. I’ll merge when it eventually turns green

**Pull request merged**